src/api/config.c

```diff
@@ -612,6 +612,53 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr, int ParamMaxLength)
+{
+    FILE *fPtr;
+    int foundSection = 0;
+    if (ParamPtr == NULL || SectionName == NULL || ParamName == NULL)
+        return M64ERR_INPUT_INVALID;
+    if (FileName == NULL || (fPtr = fopen(FileName, "rb")) == NULL)
+    {
+        DebugMessage(M64MSG_ERROR, "Unable to open config file '%s'.", FileName);
+        return M64ERR_INPUT_INVALID;
+    }
+
+    char *buffer = malloc(ParamMaxLength + 64); //ParamMaxLength is the size of the value, we add 64 to make sure we have enough room to store the rest of the line.
+
+    while (fgets(buffer, sizeof(buffer), fPtr) != NULL)
```

Comment:
No problem, I don't want to cause any memory leaks or anything. I've added a commit to have it work like it does in ConfigInit, it does look like a much better way to do it.

I've tested it with reading GLideN64.custom.ini and it still works so that's a good sign :)